### PR TITLE
Avoid logging exceptions produced by validation errors of import/export

### PR DESCRIPTION
**Pull request by fluiddot**
Related to 8626-gh-Automattic/dotcom-forge.

## Proposed Changes

- Avoid throwing an exception when the import/export process encounters a validation error.
- Remove console log upon SQL file processing errors, rely on the common error handling and reporting to Sentry.
- Avoid logging to console backup import failures.

## Testing Instructions

**Preparation:**
- Apply the following patch to observe generation of Sentry events:
```patch
diff --git forkSrcPrefix/src/index.ts forkDstPrefix/src/index.ts
index 34ff7a348be5596afcff44de012606a08881d028..3f33e7976bf97f06a2390c427acd9ce1db36a6d2 100644
--- forkSrcPrefix/src/index.ts
+++ forkDstPrefix/src/index.ts
@@ -39,11 +39,12 @@ if ( ! isCLI() ) {
 	Sentry.init( {
 		dsn: 'https://97693275b2716fb95048c6d12f4318cf@o248881.ingest.sentry.io/4506612776501248',
 		debug: true,
-		enabled:
-			process.env.NODE_ENV !== 'development' &&
-			process.env.NODE_ENV !== 'test' &&
-			! process.env.E2E,
+		enabled: true,
 		release: `${ app.getVersion() ? app.getVersion() : COMMIT_HASH }-${ getPlatformName() }`,
+		beforeSend: ( event ) => {
+			console.error( event.exception?.values );
+			return null; // this drops the event and nothing will be sent to sentry
+		},
 	} );
 }
 

```

### Valid importer not found

- Run the app with the command `STUDIO_IMPORT_EXPORT=true npm start`.
- Select a site and navigate to the Import/Export tab.
- Import a file with a non-supported extension (e.g. text file).
- Observe the app shows an error.
- **Observe in the logs that no Sentry log has been generated.**

### Error processing SQL

- Apply the following patch to force an error in processing:
```patch
diff --git forkSrcPrefix/src/site-server.ts forkDstPrefix/src/site-server.ts
index 5aa86c43b350ce7d243840e15388af87c2732888..5577b67c2f5e1003b269a7af4888524a1b22f516 100644
--- forkSrcPrefix/src/site-server.ts
+++ forkDstPrefix/src/site-server.ts
@@ -161,6 +161,9 @@ export class SiteServer {
 	}
 
 	async executeWpCliCommand( args: string ): Promise< WpCliResult > {
+		if ( args.startsWith( 'sqlite import' ) ) {
+			throw Error( 'testing' );
+		}
 		const projectPath = this.details.path;
 
 		if ( ! this.wpCliExecutor ) {

```
- Run the app with the command `STUDIO_IMPORT_EXPORT=true npm start`.
- Select a site and navigate to the Import/Export tab.
- Import an SQL file.
- Observe the app shows an error.
- **Observe in the logs that a Sentry log has been generated.**

### Avoid logging import failures twice

- Apply the following patch to force an exception during import:
```patch
diff --git forkSrcPrefix/src/lib/import-export/import/importers/importer.ts forkDstPrefix/src/lib/import-export/import/importers/importer.ts
index b8baea7ca850e70d02af02e2ad18274b9f36c63a..46c1bbd1e77f554d99c8649794d17d6940b882d1 100644
--- forkSrcPrefix/src/lib/import-export/import/importers/importer.ts
+++ forkDstPrefix/src/lib/import-export/import/importers/importer.ts
@@ -80,6 +80,7 @@ export class JetpackImporter extends BaseImporter {
 		this.emit( ImportEvents.IMPORT_START );
 
 		try {
+			throw new Error( 'testing' );
 			const databaseDir = path.join( rootPath, 'wp-content', 'database' );
 			const dbPath = path.join( databaseDir, '.ht.sqlite' );
 

```
- Run the app with the command `STUDIO_IMPORT_EXPORT=true npm start`.
- Select a site and navigate to the Import/Export tab.
- Import a Jetpack backup.
- Observe the app shows an error.
- **Observe in the logs that a Sentry log has been generated.**
- Observe no other error messages are logged.

### Valid exporter not found

- The app doesn't allow using an invalid extension when exporting, so we need to apply the following patch:
```patch
diff --git forkSrcPrefix/src/lib/import-export/export/exporters/default-exporter.ts forkDstPrefix/src/lib/import-export/export/exporters/default-exporter.ts
index 2645d6fc201497e94d2027a0eaead04bf5e85cf7..c8575032e57f7ec8dcf4f17bf716d66610879982 100644
--- forkSrcPrefix/src/lib/import-export/export/exporters/default-exporter.ts
+++ forkDstPrefix/src/lib/import-export/export/exporters/default-exporter.ts
@@ -37,6 +37,7 @@ export class DefaultExporter extends EventEmitter implements Exporter {
 		};
 	}
 	async canHandle(): Promise< boolean > {
+		return false;
 		// Check for supported extension
 		const supportedExtension = [ 'tar.gz', 'tzg', 'zip' ].find( ( ext ) =>
 			this.options.backupFile.endsWith( ext )

```
- Run the app with the command `STUDIO_IMPORT_EXPORT=true npm start`.
- Select a site and navigate to the Import/Export tab.
- Export a site.
- Observe the app shows an error.
- **Observe in the logs that no Sentry log has been generated.**

## Pre-merge Checklist

- [x] Have you checked for TypeScript, React or other console errors?

### CODE REVIEW

File: src/lib/import-export/import/handlers/backup-handler-factory.ts [hunk 1]
```diff
@@ -47,15 +47,13 @@ export class BackupHandlerFactory {
...
-	static create( file: BackupArchiveInfo ): BackupHandler {
+	static create( file: BackupArchiveInfo ): BackupHandler | undefined {
 		if ( this.isZip( file ) ) {
 			return new BackupHandlerZip();
 		} else if ( this.isTarGz( file ) ) {
 			return new BackupHandlerTarGz();
 		} else if ( this.isSql( file ) ) {
 			return new BackupHandlerSql();
-		} else {
-			throw new Error( 'Unsupported file format. Only zip, tar.gz, and sql files are supported.' );
```
Comment (sejas) [inline src/lib/import-export/import/handlers/backup-handler-factory.ts:58]: Nice!

File: src/lib/import-export/import/import-manager.ts [hunk 2]
```diff
@@ -57,11 +60,8 @@ export async function importBackup(
 			await backupHandler.extractFiles( backupFile, extractionDirectory );
 			return await importer.import( site.path, site.id );
 		} else {
-			throw new Error( 'No suitable importer found for the given backup file' );
+			return;
 		}
```
Comment (sejas) [inline src/lib/import-export/import/import-manager.ts]: Should we write the inverse check?
```
if ( ! importer ) {
   return;
}
```

**Review by sejas — APPROVED** [review]
I tested the import and export flows by using the Diffs provided, and I confirm they work as expected.

`Valid importer not found`, `Error processing SQL` and `Avoid logging import failures twice`

https://github.com/user-attachments/assets/4ff00875-72f6-4f34-8694-c7f4c8981503

And the export flow: `Valid exporter not found`

https://github.com/user-attachments/assets/cc16ba9e-f334-4b89-9383-be0dc1aec0e4

File: src/lib/import-export/import/import-manager.ts (hunk 2 above)
Comment (fluiddot) [reply src/lib/import-export/import/import-manager.ts]: Yep, I agree, let's change it.

Comment (fluiddot) [reply src/lib/import-export/import/import-manager.ts]: I've applied this suggestion in https://github.com/Automattic/studio/pull/453/commits/fa9731b515e1bdcdff53140c0467c361e344b851.

**Pull request merged**